codersdk/deployment.go

```diff
@@ -942,6 +951,33 @@ when required by your organization's security policy.`,
 			Group:       &deploymentGroupIntrospectionPrometheus,
 			YAML:        "collect_agent_stats",
 		},
+		{
+			Name:        "Prometheus Aggregate Agent Stats By",
+			Description: fmt.Sprintf("When collecting agent stats, aggregate metrics by a given set of comma-separated labels to reduce cardinality. Accepted values are %s.", strings.Join(AcceptedMetricAggregationLabels, ", ")),
+			Flag:        "prometheus-aggregate-agent-stats-by",
+			Env:         "CODER_PROMETHEUS_AGGREGATE_AGENT_STATS_BY",
+			Value: clibase.Validate(&c.Prometheus.AggregateAgentStatsBy, func(value *clibase.StringArray) error {
```

Comment:
Would it be beneficial to extract this function outside and have it tested?

Reply:
👍 probably no harm to do this